src/mlpack/methods/det/dtree_impl.hpp

```diff
@@ -165,9 +165,138 @@ DTree<MatType, TagType>::DTree() :
     right(NULL)
 { /* Nothing to do. */ }
 
+template <typename MatType, typename TagType>
+DTree<MatType, TagType>::DTree(const DTree& obj) :
+    start(obj.start),
+    end(obj.end),
+    maxVals(obj.maxVals),
+    minVals(obj.minVals),
+    splitDim(obj.splitDim),
+    splitValue(obj.splitValue),
+    logNegError(obj.logNegError),
+    subtreeLeavesLogNegError(obj.subtreeLeavesLogNegError),
+    subtreeLeaves(obj.subtreeLeaves),
+    root(obj.root),
+    ratio(obj.ratio),
+    logVolume(obj.logVolume),
+    bucketTag(obj.bucketTag),
+    alphaUpper(obj.alphaUpper),
+    left(obj.left),
+    right(obj.right)
+{ /* Nothing to do. */ }
```

Comment:
This is the copy constructor, so you need to copy the data. Take a look at the copy constructor test you wrote---if you just copy the pointer value, then when you delete the tree you copied from, suddenly the left and right pointers point to invalid memory. I bet that if you run valgrind to check for memory leaks on that test, like

```
$ bin/mlpack_test -t DETTest/CopyConstructorAndOperatorTest
```

then valgrind will throw invalid memory access errors everywhere, even if the test itself does not fail.